### ISSUE TITLE
Allow qml.draw_mpl to draw onto a specified figure or axis

### ISSUE THREAD

**Issue by eendebakpt** [PennyLaneAI/pennylane#4950]
### Feature details

The current implementation of `qml.draw_mpl` always opens a new figure window. It would be nice if we can pass a matplotlib figure or axis, e.g.
```
qml.draw_mpl(circuit, fig=plt.gcf())
```
or
```
plt.figure()
ax = plt.subplot(1,3, 1)
qml.draw_mpl(circuit, ax = ax)
...
```

### Implementation

Typically this is implemented by passing either the figure or axis argument to the low-level drawing functionality. 

### How important would you say this feature is?

1: Not important. Would be nice to have.

**Comment by josh146** [comment]
Thanks @eendebakpt! This would be a nice improvement to the Drawer functionality. Would you be open to making a PR to add this functionality?

**Comment by albi3ro** [comment]
I'm going to mark this closed by PR #4956
